### PR TITLE
Yandex Geocode Support

### PR DESCRIPTION
**Pull request by dryuri92**
add yandex Geocode API

### CODE REVIEW

File: services/geocode/geocode.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	s.parameters = fmt.Sprintf("%s&results=%d", s.parameters, limit)
+}
+
+func (s *GeocodeAPI) AddBoundaryBox(minlng, minlat, maxlng, maxlat float64) {
+	s.parameters = fmt.Sprintf("%s&bbox=%.6f,%.6f~%.6f,%.6f", s.parameters, minlng, minlat, maxlng, maxlng)
+}
+
+func (s *GeocodeAPI) ForwardGeocodeURL(address string) string {
+	URL := string(s.endpoint) + s.parameters + "&geocode=" + address
+	s.parameters = ""
```
Comment (aleksey-korolev) [inline services/geocode/geocode.go]: this approach will make it impossible to use same geocoder instance in parallel

Comment (dryuri92) [reply services/geocode/geocode.go]: Modify

File: services/suggest/suggest.go [hunk 2]
```diff
@@ -32,18 +32,25 @@ func getURL(apiKey string, baseURLs ...string) string {
...
+		parameters: fmt.Sprintf("%s&results=%d", s.parameters, limit)}
 }
 
+func (s SuggestAPI) AddBoundaryBox(minlng, minlat, maxlng, maxlat float64) SuggestAPI {
+	return SuggestAPI{endpoint: s.endpoint,
+		parameters: fmt.Sprintf("%s&bbox=%.6f,%.6f~%.6f,%.6f", s.parameters, minlng, minlat, maxlng, maxlng)}
+}
 func (s *SuggestAPI) GeosuggestURL(address string) string {
 	URL := string(s.endpoint) + s.parameters + "&text=" + address
 	s.parameters = ""
```
Comment (aleksey-korolev) [inline services/suggest/suggest.go]: building url has side effect of cleaning parameters

File: services/geocode/geocode.go [hunk 3]
```diff
@@ -18,8 +18,8 @@ type (
 	}
 )
 
-func NewGeocodeAPI(apiKey string, baseURLs ...string) *GeocodeAPI {
-	return &GeocodeAPI{
+func NewGeocodeAPI(apiKey string, baseURLs ...string) GeocodeAPI {
+	return GeocodeAPI{
```
Comment (aleksey-korolev) [inline services/geocode/geocode.go:22]: why is thos approach better than using builder that used to be used in the lib?

File: example/main.go [hunk 4]
```diff
@@ -11,32 +14,48 @@ import (
...
 	if err != nil {
 		log.Fatalf("Suggest error: %v", err)
 	}
 	fmt.Printf("Suggest for text is %v\n", suggestion)
 	fmt.Printf("Suggested len are: %d\n", len(suggestion.Results))
 	fmt.Printf("Suggested 0 result are: %s\n", suggestion.Results[0].Title.Text)
+	geocodeAPI := geocode.NewGeocodeAPI(apiGeocoder)
+	geocodeAPI = geocodeAPI.AddLanguage(lang).
+		AddLimit(limit)
+	geocodeAPI.AddLimit(limit)
```
Comment (aleksey-korolev) [inline example/main.go]: adds limit twice

Comment (dryuri92) [reply example/main.go]: Done

**Pull request merged**